Applications/Utils/MeshEdit/removeMeshElements.cpp

```diff
@@ -11,35 +11,44 @@
  *              http://www.opengeosys.org/LICENSE.txt
  */
 
+#include <memory>
+
 // TCLAP
 #include "tclap/CmdLine.h"
 
-// ThirdParty/logog
-#include "logog/include/logog.hpp"
-
-// BaseLib
-#include "LogogSimpleFormatter.h"
+#include "Applications/ApplicationsLib/LogogSetup.h"
 
-// FileIO
-#include "Legacy/MeshIO.h"
-#include "readMeshFromFile.h"
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/writeMeshToFile.h"
 
-// MeshLib
-#include "Mesh.h"
+#include "MeshLib/Mesh.h"
 #include "MeshLib/Node.h"
-#include "Elements/Element.h"
-#include "MeshEnums.h"
-#include "MeshSearch/ElementSearch.h"
-#include "MeshEditing/RemoveMeshComponents.h"
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/MeshEnums.h"
+#include "MeshLib/MeshSearch/ElementSearch.h"
+#include "MeshLib/MeshEditing/RemoveMeshComponents.h"
+
+template <typename PROPERTY_TYPE>
+void searchByProperty(std::string const& property_name,
+                      std::vector<PROPERTY_TYPE> const& property_values,
+                      MeshLib::ElementSearch& searcher)
+{
+	for (auto property_value : property_values) {
+		const std::size_t n_removed_elements =
+		    searcher.searchByPropertyValue(property_value, property_name);
+		INFO("%d elements with property value %d found.", n_removed_elements,
```

Comment:
`%d` is problematic since `std::size_t` can be `unsigned long`, then one would need `%lu` in a strict sense. However, that problem is also present in many other locations. Ignore it here, but we need some type-safe logger.
One could workaround this by writing
```
INFO("...%s ...", std::to_string(n_removed_elements));
```
But that's ugly.